Build fails on clang 3.9, 4 & 5 (not 6)

I've created a [conan package for aeron](https://github.com/helmesjo/conan-aeron). It [fails when building](https://travis-ci.org/helmesjo/conan-aeron/builds/417223976) with clang (except version 6) with the following error:
```
aeron-client/src/main/cpp/util/MemoryMappedFile.h:42:50: error: unknown type name 'off_t'
    static ptr_t createNew(const char* filename, off_t offset, size_t length)
```

PR #315 changed usages of `size_t` to `off_t`. Was this really necessary to make it compile for windows?

Any suggestions for a workaround?

--- a/aeron-client/src/main/cpp/util/MemoryMappedFile.cpp
+++ b/aeron-client/src/main/cpp/util/MemoryMappedFile.cpp
@@ -16,7 +16,6 @@
 
 #ifndef _WIN32
     #include <sys/mman.h>
-    #include <sys/types.h>
     #include <sys/stat.h>
     #include <fcntl.h>
     #include <unistd.h>

--- a/aeron-client/src/main/cpp/util/MemoryMappedFile.h
+++ b/aeron-client/src/main/cpp/util/MemoryMappedFile.h
@@ -20,11 +20,12 @@
 #include <memory>
 
 #ifdef _WIN32
-#ifndef NOMINMAX
-#define NOMINMAX
-#endif // !NOMINMAX
-
-#include <windows.h>
+    #ifndef NOMINMAX
+        #define NOMINMAX
+    #endif // !NOMINMAX
+    #include <windows.h>
+#else
+    #include <sys/types.h>
 #endif
 
 

